crates/swc_ecma_transforms/tests/es2015_function_name.rs

```diff
@@ -57,7 +57,7 @@ test!(
     r#"number = function (x) {
   return x;
 };"#,
-    r#"number = function number(x) {
+    r#"number = function number1(x) {
```

Comment:
@kdy1 

After I used `private_ident!`, the problem of #2753 is solved. But the change in this test case is unexpected.

Hygiene Renamer would rename `number#private` to `number1` for these is an unresolved `number` in the global scope.

I wonder if Hygiene Renamer needs to apply special behaviors on top-level and non-top-level variables.

Reply:
I think the ideal behavior is to check if there's a reference for `number#global` in the scope of `number#private` and rename only if there are usages for both, but I'm not sure how we can implement it.

Reply:
Do you consider these test changes unacceptable? If not, we could merge this PR first after the CI is passed.

Reply:
I think it's fine